src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.GCDesc.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    partial class EETypeNode
+    {
+        private ISymbolNode _gcdesc;
+
+        private ISymbolNode GCDesc
+        {
+            get
+            {
+                if (GCDescSize == 0)
+                    return null;
+                if (_gcdesc == null)
+                {
+                    System.Threading.Interlocked.CompareExchange(ref _gcdesc,
+                        new ObjectAndOffsetSymbolNode(this, 0, "__GCDesc_" + NodeFactory.NameMangler.GetMangledTypeName(_type)), null);
+                }
+                return _gcdesc;
+            }
+        }
+
+        private int GCDescSize
+        {
+            get
+            {
+                if (!_constructed || _type.IsGenericDefinition)
+                    return 0;
+
+                GCPointerMap map = GetGCPointerMap();
+                if (!map.IsInitialized)
+                    return 0;
+
+                int numSeries = 0;
+                for (int i = 0; i < map.Size; i++)
+                {
+                    if (map[i])
+                    {
+                        numSeries++;
+                        while (++i < map.Size && map[i]) ;
+                    }
+                }
+
+                if (_type.IsArray || _type.IsSzArray)
+                {
+                    return numSeries > 0 ? (numSeries + 2) * _type.Context.Target.PointerSize : 0;
+                }
+                else
+                {
+                    return numSeries > 0 ? (numSeries * 2 + 1) * _type.Context.Target.PointerSize : 0;
+                }
+            }
+        }
+
+        private GCPointerMap GetGCPointerMap()
+        {
+            switch (_type.Category)
+            {
+                case TypeFlags.SzArray:
+                case TypeFlags.Array:
+                    {
+                        var elementType = ((ArrayType)_type).ElementType;
+                        if (elementType.IsObjRef)
+                        {
+                            return new GCPointerMap(new int[] { 0x1 }, 1);
```

Comment:
This is apparently wrong. These need to be special cased in some way I have yet to figure out.